core/src/main/java/org/mule/runtime/core/processor/IdempotentRedeliveryPolicy.java

```diff
@@ -78,15 +81,15 @@ public void initialise() throws InitialisationException {
       try {
         MessageDigest.getInstance(messageDigestAlgorithm);
       } catch (NoSuchAlgorithmException e) {
-        throw new InitialisationException(CoreMessages.initialisationFailure(String
-            .format("Exception '%s' initializing message digest algorithm %s", e.getMessage(), messageDigestAlgorithm)), this);
+        throw new InitialisationException(initialisationFailure(format("Exception '%s' initializing message digest algorithm %s",
+                                                                       e.getMessage(), messageDigestAlgorithm)),
+                                          this);
 
       }
     }
 
-    String appName = muleContext.getConfiguration().getId();
-    String flowName = flowConstruct.getName();
-    idrId = String.format("%s-%s-%s", appName, flowName, "idr");
+    idrId = format("%s-%s-%s", muleContext.getConfiguration().getId(),
+                   getLocation().getParts().get(0).getPartPath(), "idr");
```

Comment:
Refactor repeated use of this type of thing into static helpers in ComponentLocation.